weaviate/connection/rest.go

```diff
@@ -56,34 +56,47 @@ func NewConnection(scheme string, host string, httpClient *http.Client, headers
 
 // WaitForWeaviate waits until weaviate is started up and ready
 func (con *Connection) WaitForWeaviate(startupTimeout time.Duration) error {
-	if startupTimeout < 0 {
-		return errors.New("'startupTimeout' needs to be a time.Duration larger than zero")
-	}
-	ticker := time.NewTicker(time.Second)
-	startTime := time.Now()
-	for {
-		t := <-ticker.C
+	switch {
```

Comment:
total nitpick:
you don't need the switch here and can just do
```
if startupTimeout < 0{
	return errors.New("'startupTimeout' can not be smaller than zero")
} else if startupTimeout == 0{
	return nil
}
...rest of the code..
```

returning early for errors/easy caes and then having the "main path" at the base intendation reduces the cognitive load when reading code and is in general a good practice
([Keep code left](https://medium.com/@zacharyrsmith/keep-code-left-how-else-statements-are-hurting-your-code-39497e390b74)!)